packages/components/managed-data-table/src/data-table-settings/data-table-settings.js

```diff
@@ -0,0 +1,189 @@
+  DISPLAY_SETTINGS,
+} from '../constants';
+import messages from './messages';
+
+export const getDropdownOptions = ({
+  isColumnManagerEnabled,
+  isDisplaySettingsEnabled,
+  formatMessage,
+}) => [
+  ...(isColumnManagerEnabled
+    ? [
+        {
+          value: COLUMN_MANAGER,
+          label: formatMessage(messages.columnManagerOption),
+        },
+      ]
+    : []),
+  ...(isDisplaySettingsEnabled
+    ? [
+        {
+          value: DISPLAY_SETTINGS,
+          label: formatMessage(messages.displaySettingsOption),
+        },
+      ]
+    : []),
+];
+
+export const getMappedColumns = (columns = []) =>
+  columns.reduce(
+    (mappedColumns, column) => ({
+      ...mappedColumns,
+      [column.key]: column,
+    }),
+    {}
+  );
+
+export const getSelectedColumns = (visibleColumnsKeys = [], mappedColumns) =>
+  visibleColumnsKeys.map((columnKey) => mappedColumns[columnKey]);
+
+const DataTableSettings = (props) => {
+  const intl = useIntl();
+  const [openPanel, setOpenPanel] = useState(null);
+
+  const dropdownOptions = getDropdownOptions({
+    isColumnManagerEnabled: !props.columnManager.disableColumnManager,
+    isDisplaySettingsEnabled: !props.displaySettings.disableDisplaySettings,
+    formatMessage: intl.formatMessage,
+  });
+
+  const handleDropdownChange = (event) => {
+    if (event.target.value === COLUMN_MANAGER) {
+      setOpenPanel(COLUMN_MANAGER);
+      return;
+    }
+    setOpenPanel(DISPLAY_SETTINGS);
+  };
+
+  const mappedColumns = getMappedColumns(props.columnManager.hideableColumns);
+
+  const selectedColumns = getSelectedColumns(
+    props.columnManager.visibleColumnKeys,
+    mappedColumns
+  );
+
+  return (
+    <Spacings.Stack>
+      <Spacings.Inline justifyContent="space-between" alignItems="center">
+        {props.topBar || <div />}
+        {dropdownOptions.length > 0 && (
+          <Constraints.Horizontal constraint="s">
+            <SelectInput
+              name="table-settings-dropdown"
+              value=""
+              placeholder={intl.formatMessage(messages.placeholder)}
+              onChange={handleDropdownChange}
+              options={dropdownOptions}
+            />
+          </Constraints.Horizontal>
+        )}
+      </Spacings.Inline>
+      {openPanel === DISPLAY_SETTINGS && (
+        <DensityManager
+          data-testid={DISPLAY_SETTINGS}
+          {...props.displaySettings}
+          isDensityManagerOpenDefault={true}
+          onClose={() => {
+            setOpenPanel(null);
+          }}
+          onDensityDisplayChange={() => {
+            props.onSettingsChange(
+              TABLE_SETTINGS_ACTIONS.IS_TABLE_CONDENSED_UPDATE,
+              !props.displaySettings.isCondensed
+            );
+          }}
+          onTextWrappingChange={() => {
+            props.onSettingsChange(
+              TABLE_SETTINGS_ACTIONS.IS_TABLE_WRAPPING_TEXT_UPDATE,
+              !props.displaySettings.isWrappingText
+            );
+          }}
```

Comment:
Here as well, you might want to assign those callback functions to variables and wrap them with `React.useCallback`

Reply:
Specially in this ones where the next value is toggling the current one for the prop, is this necessary?

Reply:
It's not always necessary, it mostly depends what the underlying component does with the prop